logger/common.go

```diff
@@ -178,7 +187,7 @@ func (l *Logger) read(s *bufio.Scanner, source string) error {
 	if debug.Verbose {
```

Comment:
Looks like I can't comment on line 183....so I put my comment here: the scanner can only set its error in `scanner.Scan()`, and this error has been handled in line 165. When we enter the for loop, the `scanner.Scan()` must be true without an error. So I don't think we need to check `s.Err()` in line 183. 

This is under the assumption that the `scan.go` follows the best practice: when it returns `true`, it will not set `Err()`. We can of course still keep this error check, if we assumes that later the go library may change its behavior.

Reply:
I think it's better to keep it for now just in case go library change this behavior for now.